spacy_llm/pipeline/llm.py

```diff
@@ -134,17 +145,30 @@ def pipe(self, stream: Iterable[Doc], *, batch_size: int = 128) -> Iterator[Doc]
             ]
             try:
                 prompts = self._task.generate_prompts(noncached_doc_batch)
+                if self._save_io:
+                    prompts, saved_prompts = tee(prompts)
```

Comment:
I don't think `itertools.tee` helps here since we consume the iterator in the backend but I've never properly profiled this so I'm not sure

Reply:
Performance isn't that critical here anyway, I believe.

Reply:
Exactly. See [comment](https://github.com/explosion/spacy-llm/pull/127#discussion_r1206379607).